staging_docs/user/guides/publish.md

```diff
@@ -102,10 +102,14 @@ $ http $BASE_ADDR/pypi/foo/simple/
 $ http $BASE_ADDR/pypi/foo/simple/shelf-reader/
 ```
 
+!!! note
+    When domains are enabled, it is necessary to include the domain name within the URL, like so:
+    `$BASE_ADDR/pypi/${DOMAIN_NAME}/foo/simple/`
+
 The content is also pip installable:
 
 ```bash
-$ pip install --trusted-host localhost -i $BASE_ADDR/pypi/foo/simple/ shelf-reader
+$ pip install --trusted-host localhost:$PORT -i $BASE_ADDR/pypi/foo/simple/ shelf-reader --no-build-isolation
```

Comment:
How is no-build-isolation better in any way? If it were the silver bullet, it would be on by default. 
In any way, I don't think we should teach people about this flag. It is completely orthogonal to how the package is delivered, which is the topic of _this_ documentation.
Which brings me to the question if you could configure a virtual environment to use a certain server.

Reply:
In fact:
https://pip.pypa.io/en/stable/topics/configuration/

Reply:
I was running these snippets in my virtual env. Using this flag helped me to resolve the issues with the installation. If this cannot be approved, we may think about using a different package instead of `shelf-reader`. Is there any good replacement for it?

Reply:
In fact, whether you need to turn of the build isolation is a question of the package you want to install.

On what criteria was shelf-reader chosen? Does it have "no dependencies"?

Reply:
Can you try installing shelf-reader again? Nightly is green again, so I guess the latest version of setuptools fixed the issue.

Reply:
Looks like the error still persists:
```
Successfully installed setuptools-71.1.0
(venv3) [lmjachky@lmjachky-thinkpadt14gen4 pulpcore]$ pip --no-cache-dir install --trusted-host localhost:5001 -i $BASE_ADDR/pypi/default/foo/simple/ shelf-reader
Looking in indexes: http://localhost:5001/pypi/default/foo/simple/
Collecting shelf-reader
  Downloading http://localhost:5001/pulp/content/default/foo/shelf-reader-0.1.tar.gz (19 kB)
  Installing build dependencies ... error
  error: subprocess-exited-with-error
  
  × pip subprocess to install build dependencies did not run successfully.
  │ exit code: 1
  ╰─> [3 lines of output]
      Looking in indexes: http://localhost:5001/pypi/default/foo/simple/
      ERROR: Could not find a version that satisfies the requirement setuptools>=40.8.0 (from versions: none)
      ERROR: No matching distribution found for setuptools>=40.8.0
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× pip subprocess to install build dependencies did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

Reply:
The issue still persists.

```
Successfully installed setuptools-71.1.0
(venv-test) [lmjachky@lmjachky-thinkpadt14gen4 pulpcore]$ rm -fr ~/.cache/pip/
(venv-test) [lmjachky@lmjachky-thinkpadt14gen4 pulpcore]$ pip3 uninstall shelf-reader
Found existing installation: shelf-reader 0.1
Uninstalling shelf-reader-0.1:
  Would remove:
    /home/lmjachky/Workplace/pulp/pulpcore/venv-test/bin/shelf-reader
    /home/lmjachky/Workplace/pulp/pulpcore/venv-test/lib/python3.12/site-packages/example/*
    /home/lmjachky/Workplace/pulp/pulpcore/venv-test/lib/python3.12/site-packages/shelf_reader-0.1.dist-info/*
    /home/lmjachky/Workplace/pulp/pulpcore/venv-test/lib/python3.12/site-packages/shelf_reader/*
    /home/lmjachky/Workplace/pulp/pulpcore/venv-test/lib/python3.12/site-packages/tests/*
Proceed (Y/n)? y
  Successfully uninstalled shelf-reader-0.1
(venv-test) [lmjachky@lmjachky-thinkpadt14gen4 pulpcore]$ pip3 install --trusted-host localhost:5001 -i $BASE_ADDR/pypi/default/foo/simple/ shelf-reader
Looking in indexes: http://localhost:5001/pypi/default/foo/simple/
Collecting shelf-reader
  Downloading http://localhost:5001/pulp/content/default/foo/shelf-reader-0.1.tar.gz (19 kB)
  Installing build dependencies ... error
  error: subprocess-exited-with-error
  
  × pip subprocess to install build dependencies did not run successfully.
  │ exit code: 1
  ╰─> [3 lines of output]
      Looking in indexes: http://localhost:5001/pypi/default/foo/simple/
      ERROR: Could not find a version that satisfies the requirement setuptools>=40.8.0 (from versions: none)
      ERROR: No matching distribution found for setuptools>=40.8.0
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× pip subprocess to install build dependencies did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
```

Reply:
I was using a fresh python environment. If the issue cannot be reproduced on your side, I can revert the change that adds `--no-build-isolation`. The `$PORT` variable seems not to be required after testing it once again. But, I would like to keep the note about domains untouched.